addons/pos_sale/report/sale_report.py

```diff
@@ -29,10 +24,10 @@ def _query(self, with_clause='', fields={}, groupby='', from_clause=''):
             sum(l.qty * u.factor) AS qty_delivered,
             CASE WHEN pos.state = 'invoiced' THEN sum(qty) ELSE 0 END AS qty_invoiced,
             CASE WHEN pos.state != 'invoiced' THEN sum(qty) ELSE 0 END AS qty_to_invoice,
-            (COALESCE(COALESCE((l.price_unit * l.qty * u.factor * ((1 - COALESCE(l.discount / 100, 0)) )) + (l.price_unit * l.qty * u.factor * ((1 - COALESCE(l.discount / 100, 0)) ) * pol_tax.amount), (l.price_unit * l.qty * u.factor * ((1 - COALESCE(l.discount / 100, 0)) ))), 1.0)) / MIN(CASE COALESCE(pos.currency_rate, 0) WHEN 0 THEN 1.0 ELSE pos.currency_rate END) AS price_total,
+            (COALESCE(COALESCE(l.price_subtotal_incl, (l.price_unit * l.qty * u.factor * ((1 - COALESCE(l.discount / 100, 0)) ))), 1.0)) / MIN(CASE COALESCE(pos.currency_rate, 0) WHEN 0 THEN 1.0 ELSE pos.currency_rate END) AS price_total,
             (COALESCE((l.qty * u.factor * l.price_unit * ((1 - COALESCE(l.discount / 100, 0)) )), 1.0)) / MIN(CASE COALESCE(pos.currency_rate, 0) WHEN 0 THEN 1.0 ELSE pos.currency_rate END) AS price_subtotal,
```

Comment:
Can you change 
`(COALESCE((l.qty * u.factor * l.price_unit * ((1 - COALESCE(l.discount / 100, 0)) )), 1.0)) / MIN(CASE COALESCE(pos.currency_rate, 0) WHEN 0 THEN 1.0 ELSE pos.currency_rate END) AS price_subtotal,`

and use [field subtotal](https://github.com/odoo/odoo/blob/12.0/addons/point_of_sale/models/pos_order.py#L1025) instead, please review Issue #24746

Reply:
This does not solve the complete problem - it will just fix the price_total value - the price_subtotal value still will be wrong.
price_subtotal = Tax excluded
price_total = Tax included

Currently it does use the price_unit field for the price_subtotal calculation - but this field is tax included if you use a system with gross prices. So it will be the total gross price - and not the total net price !

So - 

price_total = sum(l.price_subtotal_incl) / MIN(CASE COALESCE(pos.currency_rate, 0) WHEN 0 THEN 1.0 ELSE pos.currency_rate END) AS price_total

and 
price_subtotal = sum(l.price_subtotal) / MIN(CASE COALESCE(pos.currency_rate, 0) WHEN 0 THEN 1.0 ELSE pos.currency_rate END) AS price_subtotal

Or similar should be the better option here !